src/main/drivers/accgyro/accgyro_spi_icm426xx.c

```diff
@@ -168,10 +171,65 @@ static aafConfig_t aafLUT42605[AAF_CONFIG_COUNT] = {  // see table in section 5.
     [AAF_CONFIG_1962HZ] = { 63, 3968,  3 }, // 995 Hz is the max cutoff on the 42605
 };
 
+#if defined(USE_GYRO_CLKIN)
+static pwmOutputPort_t pwmGyroClk = {0};
+
+static bool initExternalClock(const extDevice_t *dev)
+{
+    if (&gyro.gyroSensor1.gyroDev.dev != dev) {
+        // only gyro1 clkin supported. TODO: support two gyros, must be implemented in the next PR
+        return false;
+    }
+    const ioTag_t tag = gyroDeviceConfig(0)->clkIn;
+    const IO_t io = IOGetByTag(tag);
+
+    const timerHardware_t *timer = timerAllocate(tag, OWNER_GYRO_CLKIN, 0);
+    if (!timer) {
+        // Error handling: failed to allocate timer
+        return false;
+    }
+
+    pwmGyroClk.io = io;
+    pwmGyroClk.enabled = true;
+
+    IOInit(io, OWNER_GYRO_CLKIN, 0);
+    IOConfigGPIOAF(io, IOCFG_AF_PP, timer->alternateFunction);
+
+    uint32_t pwmFrequency = 32000;  // PWM frequency set to 32 kHz TODO: move to config: available freq 31-50kHz
+    const uint32_t clock = timerClock(timer->tim);  // Get the timer clock frequency
+
+    const uint16_t period = clock / pwmFrequency;
+
+    // Calculate duty cycle value for 50%
+    const uint16_t value = period / 2;
+
+    // Configure PWM output
+    pwmOutConfig(&pwmGyroClk.channel, timer, clock, period - 1, value - 1, 0);
+
+    // Set CCR value
+    *pwmGyroClk.channel.ccr = value - 1;
+
+    return true;
+}
+
+static void icm426xxEnableExternalClock(const extDevice_t *dev)
+{
+    if (initExternalClock(dev)) {
+        uint8_t cfg1 = spiReadRegMsk(dev, ICM426XX_INTF_CONFIG1);
```

Comment:
From ICM-42688-P, it seems that you want to set INTF_CONFIG5[2:1] = 0b10; (and it is reserved on ICM-42605)
INTF_CONFIG1[7:4] is reserved (undocumented AFSR), you probably want to set RTC_MODE (bit 2)

Can you link documentation you are using?

Reply:
@ledvinap  You are right, I looked at the datasheet more carefully: [link](https://invensense.tdk.com/wp-content/uploads/2020/04/ds-000347_icm-42688-p-datasheet.pdf)

To activate clocking the external CLKIN signal on the ICM-42688-P, it is necessary to:
* In the INTF_CONFIG5 register (bank 1, register 0x7B), set bits 2:1 to the value 10 so that PIN9 functions as the CLKIN input.
* Set bit 2 of the RTC_MODE register (bank 0, register 0x4D) to 1, which indicates the need to use an external RTC signal.